### PR TITLE
Chap with env

### PR DESCRIPTION
**Pull request by sijeesh**

### CODE REVIEW

**Comment by AnushaY1916** [comment]
If the prehook job fails, is it logged somewhere?

File: helm/charts/hpe-csi-driver/templates/_helpers.tpl [hunk 1]
```diff
@@ -30,3 +30,38 @@ Create chart name and version as used by the chart label.
...
+
+  {{- $username := index $secret.data "chapUser" | b64dec }}
+  {{- $password := index $secret.data "chapPassword" | b64dec }}
+
+  {{- if or (empty $username) (empty $password) }}
+    {{- fail "Username or password cannot be empty." }}
+  {{- end }}
+
+  {{- $chapUserValidationPattern := "^[a-zA-Z0-9][a-zA-Z0-9\\-:.]{0,63}$" }}
+  {{- $chapPasswordValidationPattern := "^$|^[a-zA-Z0-9+_)(*^%$#@!]{12,16}$" }}
```
Comment (datamattsson) [inline helm/charts/hpe-csi-driver/templates/_helpers.tpl:52]: Isn't this regex a bit too strict? I looked at the Nimble REST API docs: 

```
CHAP secret.The CHAP secret should be between 12-16 characters and cannot contain spaces or most punctuation. String of 12 to 16 printable ASCII characters excluding ampersand and ^[];`. Example: 'password_25-24
```

Comment (sijeesh) [reply helm/charts/hpe-csi-driver/templates/_helpers.tpl:52]: This is the same pattern we used earlier for validating the secret - https://github.com/hpe-storage/co-deployments/blob/cfe36d87211e34d6ca9f77690d24fb98a887922a/helm/charts/hpe-csi-driver/values.schema.json

**Comment by sijeesh** [comment]
> Is logging required for the prehook job to know the status?

Will remove "hook-failed" from the delete-policy to debug if anything goes wrong

File: helm/charts/hpe-csi-driver/templates/_helpers.tpl (hunk 1 above)
Comment (datamattsson) [reply helm/charts/hpe-csi-driver/templates/_helpers.tpl:52]: Old code is old sins. I'm just saying that the CHAP password we use as an example in the platform docs should work with the CSI driver.

Comment (sijeesh) [reply helm/charts/hpe-csi-driver/templates/_helpers.tpl]: Agreed, there were other issues I found when I reviewed this in detail. Thank you for checking. I've updated the regex to match the description and made it more readable.

**Comment by sijeesh** [comment]
> > Is logging required for the prehook job to know the status?
> 
> Will remove "hook-failed" from the delete-policy to debug if anything goes wrong

Updated

File: operators/hpe-csi-operator/sources/hpe-csi-operator.csv.yaml [hunk 2]
```diff
@@ -157,6 +157,14 @@ spec:
...
             x-descriptors:
               - "urn:alm:descriptor:com.tectonic.ui:text"
+          - displayName: iSCSI Configuration
+            description: "Configuration for iSCSI CHAP authenticaton"
+            path: iscsi
+          - displayName: iSCSI CHAP Secret Name
+            description: "Name of the secret containing chapUser and chapPassword for iSCSI"
+            path: iscsi.chapSecretName
+            x-descriptors:
+              - "urn:alm:descriptor:com.tectonic.ui:text"
```
Comment (datamattsson) [inline operators/hpe-csi-operator/sources/hpe-csi-operator.csv.yaml:167]: Can you use `urn:alm:descriptor:io.kubernetes:Secret` here instead?

Comment (sijeesh) [reply operators/hpe-csi-operator/sources/hpe-csi-operator.csv.yaml]: Done

**Review by datamattsson — APPROVED** [review]
LGTM

**Pull request merged**